src/lib/BUILD.gn

```diff
@@ -53,10 +54,31 @@ static_library("lib") {
   }
 
   cflags = [ "-Wconversion" ]
+}
+
+static_library("lib") {
+  public_deps = [
+    ":base",
+    "${chip_root}/src/platform/logging:default",
+  ]
 
   output_name = "libCHIP"
 
   output_dir = "${root_out_dir}/lib"
 
   complete_static_lib = true
 }
+
+# This version of libCHIP is only meant for internal use by some test drivers
+static_library("lib_using_stdio") {
```

Comment:
naming wise typically `-` are used for separation. Also `lib` stutters. This has to be a build target name that looks reasonable as `"src/lib:<name>"`

how about `stdio-logging` ? then target is `"{chip_root}/src/lib:with-stdio-logging}"` which seems reasonable.

Reply:
Would it make sense to have `src/lib:without-logging` as a `static_library`, and let the executable just depend on that + the logging backend? This would make easy to also create other backends (e.g. syslog) and let that be picked via a GN arg within an app build?

Reply:
(Or maybe only stdio is special due to how we use it for tests and some tools, and for other backends the correct way would be to override a core GN arg that directly results in that backend being built into `src/lib`)

Reply:
> Or maybe only stdio is special due to how we use it for tests and some tools

@ksperling-apple, the STDIO backed is primarily for testing purposes. The "proper" logging backed should be implemented per platform basis (e.g. `syslog` in "platform/Linux/Logging.cpp"). We've been thinking about doing it in a more generalized way, so the integrator could choose logging backed. However, this will impose on the integrator to supply proper linking flags for final binary. And since the very beginning of the project there are no pkg-config of CMake helpers for the Matter SDK library, the integrator will have to "know" somehow, which libraries are required for final binary. So, in order to keep backward compatibility with current approach (there is a single `libCHIP.a` library), we decided to keep the platform logging backed in that library and create a new internal one for unit tests only (with STDIO logging).